logs/client.go

```diff
@@ -0,0 +1,199 @@
+package logs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/solarwinds/swo-cli/version"
+)
+
+type Client struct {
+	opts       *Options
+	httpClient http.Client
+	output     *os.File
+}
+
+type Log struct {
+	Time     time.Time `json:"time"`
+	Message  string    `json:"message"`
+	Hostname string    `json:"hostname"`
+	Severity string    `json:"severity"`
+	Program  string    `json:"program"`
+}
+
+type PageInfo struct {
+	PrevPage string `json:"prevPage"`
+	NextPage string `json:"nextPage"`
+}
+
+type LogsData struct {
+	Logs     []Log `json:"logs"`
+	PageInfo `json:"pageInfo"`
+}
+
+func NewClient(opts *Options) (*Client, error) {
+	return &Client{
+		httpClient: *http.DefaultClient,
+		opts:       opts,
+		output:     os.Stdout,
+	}, nil
+}
+
+func (c *Client) prepareRequest(ctx context.Context, nextPage string) (*http.Request, error) {
+	var logsEndpoint string
+	var err error
+	params := url.Values{}
+	if nextPage == "" {
+		logsEndpoint, err = url.JoinPath(c.opts.ApiUrl, "v1/logs")
+	} else {
+		u, err := url.Parse(nextPage)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse nextPage field: %w", err)
+		}
+
+		logsEndpoint, err = url.JoinPath(c.opts.ApiUrl, u.Path)
+		if err != nil {
+			return nil, err
+		}
+
+		params, err = url.ParseQuery(u.RawQuery)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	logsUrl, err := url.Parse(logsEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.opts.group != "" {
+		params.Add("group", c.opts.group)
+	}
+	if c.opts.minTime != "" {
+		params.Add("startTime", c.opts.minTime)
+	}
+	if c.opts.maxTime != "" {
+		params.Add("endTime", c.opts.maxTime)
+	}
+
+	var filter string
+	if c.opts.system != "" {
+		filter = fmt.Sprintf(`host:"%s"`, c.opts.system)
+	}
+	if len(c.opts.args) != 0 {
+		if len(filter) == 0 {
+			filter = strings.Join(c.opts.args, " ")
+		} else {
+			filter = filter + " " + strings.Join(c.opts.args, " ")
+		}
+	}
+
+	if filter != "" {
+		params.Add("filter", filter)
+	}
```

Comment:
These should all use `params.Set` instead of `params.Add`, since nextPage already includes these params.

(Ideally we wouldn't even touch params on any nextPage, since we're just replacing params with the same values–a no-op.)

Reply:
moved settings params to the if branch when nextPage is not present